src/layout/IHyprLayout.cpp

```diff
@@ -352,7 +353,7 @@ void IHyprLayout::onEndDragWindow() {
                     DRAGGINGWINDOW->m_bDraggingTiled    = false;
                 }
 
-                if (DRAGGINGWINDOW->m_sGroupData.pNextWindow.lock()) {
+                if (!DRAGGINGWINDOW->m_sGroupData.pNextWindow.expired()) {
```

Comment:
! ... .expired() redundant

if (DRAGGINGWINDOW->m_sGroupData.pNextWindow)

Reply:
Pardon my ignorance; would you mind briefly explaining me why please? (I have changed it already though).

Reply:
operator bool is just better reverse expired

Reply:
I don't understand why we don't need to check for the shared pointer here?

Reply:
May as well just leave it as .lock(), I changed it to expired() because I saw it in other parts of the code base, but as you can see I am noob at C++ memory management.

Reply:
`.lock()` was there cuz I did a bulk replace

Reply:
No, I wrote that entire block from scratch two weeks ago: https://github.com/hyprwm/Hyprland/commit/45e82199fb29f422ebbbfe06ce03022dfb6d645e. I wrote .lock() instead of .expired() by ignorance. Now that I have learnt more about Hyprland's C++ code, I think that .expired() would be better suited there, that's why I have requested this change. However, you are the expert here, now that you know the context of the situation, you decide what fits better here. Cheers.

Reply:
in hyprutils, .expired and operator bool are two different checks, but here will behave the same. That's why the shorter version is better.

`.expired()` checks if it's not possible to make a `SP` anymore.
`operator bool` checks if the pointer is still reference-able.

Difference is in the ~dtor of an object that is owned by a SP. You can't make SPs anymore (because that would resurrect the container) but you can still reference the pointer via `->` because the actual object is not deleted yet.

Reply:
Done. Thank you very much for your explanation and your patience, really appreciate them.